src/CalcManager/Ratpack/conv.cpp

```diff
@@ -300,47 +263,46 @@ PRAT numtorat(_In_ PNUMBER pin, uint32_t radix)
 //
 //----------------------------------------------------------------------------
 
-PNUMBER nRadixxtonum(_In_ PNUMBER a, uint32_t radix, int32_t precision)
+NUMBER nRadixxtonum(_In_ const NUMBER& a, uint32_t radix, int32_t precision)
 
 {
     uint32_t bitmask;
     uint32_t cdigits;
-    MANTTYPE* ptr;
+    const MANTTYPE* ptr;
 
-    PNUMBER sum = i32tonum(0, radix);
-    PNUMBER powofnRadix = i32tonum(BASEX, radix);
+    NUMBER sum = i32tonum(0, radix);
+    NUMBER powofnRadix = i32tonum(BASEX, radix);
 
     // A large penalty is paid for conversion of digits no one will see anyway.
     // limit the digits to the minimum of the existing precision or the
     // requested precision.
     cdigits = precision + 1;
-    if (cdigits > (uint32_t)a->cdigit)
+    if (cdigits > (uint32_t)a.cdigit)
     {
-        cdigits = (uint32_t)a->cdigit;
+        cdigits = (uint32_t)a.cdigit;
     }
 
     // scale by the internal base to the internal exponent offset of the LSD
-    numpowi32(&powofnRadix, a->exp + (a->cdigit - cdigits), radix, precision);
+    numpowi32(&powofnRadix, a.exp + (a.cdigit - cdigits), radix, precision);
 
     // Loop over all the relative digits from MSD to LSD
-    for (ptr = &(a->mant[a->cdigit - 1]); cdigits > 0; ptr--, cdigits--)
+    for (ptr = &(a.mant[a.cdigit - 1]); cdigits > 0; ptr--, cdigits--)
```

Comment:
The unit tests are failing, and I think the problem is here--we're calling a.mant[-1].

One reasonable fix would be to rewrite this as a while loop so we check cdigits first:

while(cdigits > 0)
{
    ptr = &(a.mant[a.cdigit - 1]);
    // etc.
    ptr--;
    cdigits--;
}